homeassistant/components/samsungtv/media_player.py

```diff
@@ -173,14 +190,16 @@ async def async_update(self) -> None:
                 STATE_ON if await self._bridge.async_is_on() else STATE_OFF
             )
 
-        if self._attr_state == STATE_ON and self._app_list is None:
-            self._app_list = {}  # Ensure that we don't update it twice in parallel
-            await self._async_update_app_list()
-
-    async def _async_update_app_list(self) -> None:
-        self._app_list = await self._bridge.async_get_app_list()
-        if self._app_list is not None:
-            self._attr_source_list.extend(self._app_list)
+        if self._attr_state == STATE_ON and not self._app_list_event.is_set():
+            await self._bridge.async_request_app_list()
```

Comment:
```suggestion
            await self._bridge.async_request_app_list()
            if self._app_list_event.is_set():
                return
```

The try+wait_for is a bit expensive so we should try not to enter it unless we have to (Python 3.11 will have zero cost try)

Reply:
Done